sql/2021/security/cookie_age_percentiles.sql

```diff
@@ -0,0 +1,105 @@
+#standardSQL
+# Percentiles of Max-Age-attribute, Expires-attribute and real age (Max-Age has precedence) of cookies.
+-- SQL Linter cannot handle DETERMINISTIC keyword so needs noqa ignore command on previous line
+CREATE TEMPORARY FUNCTION getCookieAgeValues(headers STRING, epochOfRequest NUMERIC)  -- noqa: PRS
+  RETURNS STRING DETERMINISTIC
+  LANGUAGE js AS '''
+  const regexMaxAge = new RegExp(/max-age=(?<value>-*[0-9]+)/i);
+  const regexExpires = new RegExp(/expires=(?<value>.*?)(;|$)/i);
+  const parsed_headers = JSON.parse(headers);
+  const cookies = parsed_headers.filter(h => h.name.match(/set-cookie/i));
+  const cookieValues = cookies.map(h => h.value);
+  const result = {
+      "maxAge": [],
+      "expires": [],
+      "realAge": []
+  };
+  cookieValues.forEach(cookie => {
+      let maxAge = null;
+      let expires = null;
+      if (regexMaxAge.exec(cookie)) {
+          maxAge = Number(regexMaxAge.exec(cookie)[1]);
+          result["maxAge"].push(maxAge);
+      }
+      if (regexExpires.exec(cookie)) {
+          expires = Number(Date(regexExpires.exec(cookie)[1])) - epochOfRequest;
```

Comment:
I *think* you'll want to divide `Number(Date(...))` by 1000 here. The `startedDateTime` column seems to be in seconds since epoch, while `Number(Date(...))` yields something in *milli*seconds. `maxAge` is also in seconds, so dividing here will make the numbers comparable.